src/swap/SwapExecuteScreen.tsx

```diff
@@ -118,33 +152,23 @@ export function SwapExecuteScreen() {
       case SwapState.START:
       case SwapState.APPROVE:
       case SwapState.EXECUTE:
-        return (
-          <ActivityIndicator
-            size="large"
-            color={colors.greenBrand}
-            testID="SwapExecuteScreen/loadingIcon"
-            style={styles.activityIndicator}
-          />
-        )
+        return <GreenLoadingSpinner />
       case SwapState.ERROR:
       case SwapState.PRICE_CHANGE:
-        return (
-          <View
-            testID="SwapExecuteScreen/errorIcon"
-            style={[styles.iconContainer, { backgroundColor: colors.warning }]}
-          >
-            <Times color={colors.light} />
-          </View>
-        )
+        return <RedLoadingSpinnerToInfo />
       case SwapState.COMPLETE:
-        return (
-          <View
-            testID="SwapExecuteScreen/completeIcon"
-            style={[styles.iconContainer, { backgroundColor: colors.greenUI }]}
-          >
-            <Checkmark color={colors.light} />
-          </View>
-        )
+        return <GreenLoadingSpinnerToCheck />
+    }
+  }, [swapState])
+
+  const NavigationButtons = useMemo(() => {
```

Comment:
This isn't a component, just a variable.
```suggestion
  const navigationButtons = useMemo(() => {
```